htdocs/product/class/product.class.php

```diff
@@ -2181,12 +2181,45 @@ public function fetch($id = '', $ref = '', $ref_ext = '', $barcode = '', $ignore
 		} else {
 			$sql .= " pa.accountancy_code_buy, pa.accountancy_code_buy_intra, pa.accountancy_code_buy_export, pa.accountancy_code_sell, pa.accountancy_code_sell_intra, pa.accountancy_code_sell_export,";
 		}
-		$sql .= " p.stock,p.pmp, p.datec, p.tms, p.import_key, p.entity, p.desiredstock, p.tobatch, p.batch_mask, p.fk_unit,";
+		//For MultiCompany PMP per entity
+		$separatedEntityPMP = false;
+		if (!empty($conf->global->MULTICOMPANY_PRODUCT_SHARING_ENABLED) && !empty($conf->global->MULTICOMPANY_PMP_PER_ENTITY_ENABLED)) {
+			$checkPMPPerEntity = $this->db->query("SELECT pmp FROM " . MAIN_DB_PREFIX . "entity_product_pmp WHERE fk_product  = ".((int) $id)." AND entity = ".(int) $conf->entity);
+			if ($this->db->num_rows($checkPMPPerEntity)>0) {
+				$separatedEntityPMP = true;
+			}
+		}
+		//For MultiCompany Stocks Sharings stock_reel includes only stocks shared with this entity
+		$separatedStock = false;
+		if (!empty($conf->global->MULTICOMPANY_STOCK_SHARING_ENABLED)) {
```

Comment:
To enhance readability of code, can you merge all new code neede by this new feature into the same
if (!empty($conf->global->MULTICOMPANY_STOCK_SHARING_ENABLED)) {

}

Reply:
sanitize worked
I will work on all your demands
Thanks for your remarks

Reply:
updates asked done
but Travis throws a test error for a file and a method I didn't touch ... 

1) CodingPhpTest::testPHP
Found a $this->db->idate to forge a sql request without quotes around this date field compta/facture/class/facture.class.php
Failed asserting that false is true.